openupgrade_scripts/scripts/project/15.0.1.2/post-migration.py

```diff
@@ -0,0 +1,108 @@
+from openupgradelib import openupgrade
+
+
+def _convert_field_m2o_to_m2m(env):
+    # Convert m2o to m2m in 'project.task'
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["project.task"],
+        "project_task",
+        "user_ids",
+        openupgrade.get_legacy_name("user_id"),
+    )
+
+
+def _create_missing_project_task_stage_personal(env):
+    tasks = env["project.task"].with_context(active_test=False).search([])
+    tasks._populate_missing_personal_stages()
```

Comment:
Now I understand how this goes. Again, the lack of comments and specially the whys makes this more difficult.

There's an error here in Odoo core, which is that the stages are not created in the user language. Maybe you can do such PR in upstream.

Reply:
In fact, I don't know if it's an error or not. AFAIK, the non-personal stages normally are created when the module is installed, so the name of those stages are in the lang of the user who installed the module. I don't see what should I fix. Also, if later new non-personal custom stages are added, they will appear in the language of their creator.

It's different about the personal stages, they have and user_id, and the name will appear in the language of that user_id.

Now that I think about it, it's a problem for the query if the languages are different. Maybe I should transform the query in an ORM method and handle the comparison using the correct language.